sui/tokens.js

```diff
@@ -142,7 +114,27 @@ class CoinManager {
         if (options.merge) {
             await CoinManager.mergeCoin(tx, coinTypeToCoins, options);
         } else if (options.split) {
-            await CoinManager.splitCoins(tx, coinTypeToCoins, options);
+            validateParameters({
+                isValidNumber: { split: options.split },
+            });
+
+            const coinType = options.coinType || SUI_COIN_ID;
+            const metadata = await client.getCoinMetadata({
+                coinType,
+            });
+
+            if (!metadata) {
+                printError(`No metadata found for coin type ${coinType}`);
+                process.exit(0);
+            }
+
+            const splitAmount = parseUnits(options.split, metadata.decimals);
+
+            await CoinManager.splitCoins(tx, coinTypeToCoins, {
+                ...options,
+                split: splitAmount.toString(),
+                coinType,
```

Comment:
pass these as positional args. `options` should be passed as-is, we should avoid modifying it like this, it becomes harder to reason about then (ideally an implementor can just refer to CLI options and know exactly what to expect)

It also avoids needing weird logic like converting the amount back to a string for e.g.